docs/fxa-engineering/fxa-sub-platform.md

```diff
@@ -169,13 +169,108 @@ server:
+
+The PayPal paid subscriptions are still driven by Stripe's subscription and invoicing model.  The key difference is that PayPal paid subscriptions have a [collection method](https://stripe.com/docs/api/subscriptions/object#subscription_object-collection_method) of `send_invoice`.  The [PayPal processor](#paypal-processor) is used to pay the inovices with the customer's billing agreement ID.
+
+### PayPal Accounts
+
+For development, you will need a PayPal developer account.  Once you are in the PayPal developer dashboard, navigate to "Accounts" under the Sandbox section of the menu.  Here you should create a pair of personal and business sandbox accounts.  The personal account will be use to test as the customer; the business account will be receiving the subscription payments.
+
+Once you've added the account pair, navigate to the bussines account by selecting "View/edit account".  Now click on the "API Credentials" tab.  You'll need the "NVP/SOAP Sandbox API Credentials" for the next section.
+
+### Configuration
+
+#### Auth Server
+
+In order to enable and use PayPal in the auth server, set the following configuration options
+
+```
+{
+  "subscriptions": {
+    "paypalNvpSigCredentials": true,
+    "sandbox": true,
+    "user": "your PayPal NVP API User name",
+    "pwd": "your PayPal NVP API password",
+    "signature": "your PayPal NVP API signature"
+  }
+}
+```
+
+The environment variables equivalent would be
+```
+SUBSCRIPTIONS_PAYPAL_ENABLED=true \
+PAYPAL_SANDBOX=true \
+PAYPAL_NVP_USER='your PayPal NVP API User name' \
+PAYPAL_NVP_PWD='your PayPal NVP API password' \
+PAYPAL_NVP_SIGNATURE='your PayPal NVP API signature'
+```
+
+#### Payments Server
+
+The Payments frontend also does not offer PayPal as payment provider by default.  To enable the feature, set the following configuration options
+
+```
+{
+  "featureFlags": {
+    "usePaypalUIByDefault": true,
+    }
+  },
+  "paypal": {
+    "clientId": "sb",
+    "apiUrl": "https://www.sandbox.paypal.com",
+    "scriptUrl": "https://www.paypal.com"
+  }
+}
+```
+
+Or use the environment variables
+
+```
+FEATURE_USE_SCA_PAYMENT_UI_BY_DEFAULT=true \
+PAYPAL_CLIENT_ID='sb' \
+PAYPAL_API_URL='https://www.sandbox.paypal.com' \
+PAYPAL_SCRIPT_URL='https://www.paypal.com'
+```
+
+The paypal*/PAYPAL_* values are the defaults in the repo.  For local development, you do not need to change them.
+
+### Instant Payment Notification (IPN)
+
+[PayPal IPN](https://developer.paypal.com/docs/api-basics/notifications/ipn/) is PayPal's equivalent of Stripe's webhook feature.  We do rely on IPNs in the Subscription Platform.  Unlike Stripe, however, PayPal does not offer any tool that would forward the events to your local environment.  Our team use [ngrok](https://ngrok.com/) for that purpose.
```

Comment:
Finance did recommend that developers be added to the official Mozilla PayPal account and given their own credentials to log in. Perhaps we should add a note to recommend that they use the Mozilla account for creating sandbox personal/business accounts and testing? They would need to contact the Mozilla PayPal admin in Finance to be added and they will receive an email with instructions for setting up the account.

We should also advise them to set up 2FA for this account: https://www.paypal.com/businessmanage/profile/loginSecurity, since that was a mitigation that came up in our RRA for the PayPal integration.

Reply:
Mentioned in the PayPal Accounts section.